pkg/mtls/ca.go

```diff
@@ -48,50 +48,46 @@ type ca struct {
 // Note: this function does not verify that the given certificate is actually a valid
 // StackRox service CA. To check for this, call `Validate()`.
 func LoadCAForSigning(certPEM, keyPEM []byte) (CA, error) {
-	return loadCA(certPEM, keyPEM, true)
+	ca := &ca{
+		certPEM: slices.Clone(certPEM),
+		keyPEM:  slices.Clone(keyPEM),
+	}
+
+	var err error
+	ca.tlsCert, err = tls.X509KeyPair(certPEM, keyPEM)
+	if err != nil {
+		return nil, err
+	}
+
+	ca.tlsCert.Leaf, _ = x509.ParseCertificate(ca.tlsCert.Certificate[0])
```

Comment:
Not something for this PR, but I wonder why we ignore the error here.

Reply:
Me too, I'll try to figure out if handling it breaks something.